src/CoreMatchers/Matchers/Matchers.cs

```diff
@@ -410,5 +410,81 @@ public static IMatcher<string> MatchesRegex(string value)
         {
             return new IsDictionaryWithSize<TKey, TValue>(new IsEqual<int>(size));
         }
+
+        /// <summary>
+        /// Creates a matcher for IDictionarys matching when the examined IDictionary contains at least one entry
+        /// whose key satisfies the specified keyMatcher <b>and</b> whose value satisfies the specified valueMatcher.
+        /// For example:
+        /// <code>assertThat(myMap, hasEntry(equalTo("bar"), equalTo("foo")))</code>
```

Comment:
I know I have been bad at this, but we should update the <code> part to match our actual code, so
`<code>Assert.That(myMap, HasEntry(EqualTo("bar"), EqualTo("foo")))</code>`